src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -56,6 +51,7 @@ internal AbstractApplicationBuilder(ApplicationConfiguration configuration)
         /// or setting the Agent.
         /// </summary>
         /// <param name="httpClientFactory">HTTP client factory</param>
+        /// <remarks>MSAL does not guarantee that it will not modify the HttpClient, for example by adding new headers.</remarks>
```

Comment:
Curious why we need this comment?  The headers are on a per-request basis anyway, so that shouldn't affect usage of this method.

Reply:
A customer asked about this feature (in ADAL). His questions were on the lines of - do we ever dispose the http client, do we modify it in any way? He wanted to send us his own HttpClient but in the end he decided to use 2 - one for his app, and one for ADAL.